console/program/src/data/record/entry/parse.rs

```diff
@@ -94,6 +96,44 @@ impl<N: Network> Parser for Entry<N, Plaintext<N>> {
             Ok((string, (Plaintext::Struct(IndexMap::from_iter(members.into_iter()), Default::default()), mode)))
         }
 
+        /// Parses an entry as an array: `[plaintext_0.visibility, ..., plaintext_n.visibility]`.
+        /// Observe the `visibility` is the same for all members of the plaintext value.
+        fn parse_array<N: Network>(string: &str) -> ParserResult<(Plaintext<N>, Mode)> {
+            // Parse the whitespace and comments from the string.
+            let (string, _) = Sanitizer::parse(string)?;
+            // Parse the "[" from the string.
+            let (string, _) = tag("[")(string)?;
+            // Parse the whitespace from the string.
+            let (string, _) = Sanitizer::parse_whitespaces(string)?;
+            // Parse the members.
+            let (string, (elements, mode)) = map_res(
+                separated_list1(
+                    pair(Sanitizer::parse_whitespaces, pair(tag(","), Sanitizer::parse_whitespaces)),
+                    alt((parse_literal, parse_struct, parse_array)),
+                ),
+                |members: Vec<(Plaintext<N>, Mode)>| {
+                    // Ensure the members all have the same visibility.
+                    let mode = members.iter().map(|(_, mode)| mode).dedup().collect::<Vec<_>>();
```

Comment:
are the modes ordered/sorted? if not, `dedup` won't do its job correctly

Reply:
they aren't sorted/ordered. i think the dedup here might just be a best effort attempt. you can open a PR to remove the call if its a performance hit that isn't worth paying for.